### ISSUE TITLE
Libreoffice flatpak is not respecting user theme

### ISSUE THREAD

**Issue by ahayzen** [flathub/org.libreoffice.LibreOffice#4]
What happened:
1) Install flatpak and `org.libreoffice.LibreOffice` from flathub
2) Have a theme other than Adwaita selected as the system theme (eg I have Arc-Darker) and make sure that you have the theme flatpak installed (eg i have `org.gtk.Gtk3theme.Arc-Darker`)
3) Launch the flatpak version of libreoffice
4) Observe that Adwaita is used as the theme in libreoffice

What I expected to happen:
At step 4 for Arc-Darker to be used.

As you can see in the screenshot below, left shows the flatpak version using Adwaita and right shows the system libreoffice using Arc-Darker.
![screenshot from 2017-12-02 17-28-28](https://user-images.githubusercontent.com/3669750/33518049-b27c834c-d786-11e7-9e9d-4539adfa6af6.png)

```
$ flatpak --version
Flatpak 0.10.1
$ flatpak info org.libreoffice.LibreOffice 
Ref: app/org.libreoffice.LibreOffice/x86_64/stable
ID: org.libreoffice.LibreOffice
Arch: x86_64
Branch: stable
Origin: flathub
Date: 2017-11-29 07:27:54 +0000
Subject: Allow access to org.gtk.vfs.* on the session bus (fddd567f)
Commit: 45ef1097dcad182c05e8abb90e1b68e7312ad87046bb9f0b6a2da76df501e5a5
Parent: 5283eb1de451f386984333d985d8e13528b73ee7c795e8f98eacc247b0d81a39
Location: /var/lib/flatpak/app/org.libreoffice.LibreOffice/x86_64/stable/45ef1097dcad182c05e8abb90e1b68e7312ad87046bb9f0b6a2da76df501e5a5
Installed size: 691.8 MB
Runtime: org.gnome.Platform/x86_64/3.24
```

**Comment by TingPing** [comment]
Does `flatpak run --env=SAL_USE_VCLPLUGIN=gtk3 org.libreoffice.LibreOffice` change anything?

**Comment by ahayzen** [comment]
@TingPing no this does not change anything, from what I can see it is already using the gtk3 for VCL plugin (see in the screenshot above it mentions "VCL: gtk3"). Also setting it equal to just "gtk" uses gtk2 with no theme at all, so I think this proves it is already trying to use the gtk3 VCL plugin.

**Comment by stbergmann** [comment]
Just learned that for theming to work, the app needs to have --filesystem=xdg-run/dconf, --filesystem=~/.config/dconf:ro, --talk-name=ca.desrt.dconf, --env=DCONF_USER_CONFIG_DIR=.config/dconf permissions.  The first two are not relevant for LO as it asks for full --filesystem=host anyway.  You can try to work around that for now by launching LO with

`flatpak run --talk-name=ca.desrt.dconf --env=DCONF_USER_CONFIG_DIR=.config/dconf org.libreoffice.LibreOffice`

or by doing

`flatpak override --talk-name=ca.desrt.dconf --env=DCONF_USER_CONFIG_DIR=.config/dconf org.libreoffice.LibreOffice`

one.  I'll also provide fresh builds with that fixed.

**Comment by stbergmann** [comment]
...or by doing [...] *once*

**Comment by ahayzen** [comment]
@stbergmann awesome thanks, I can confirm that `flatpak run --talk-name=ca.desrt.dconf --env=DCONF_USER_CONFIG_DIR=.config/dconf org.libreoffice.LibreOffice` resolves the issue for me :-)

**Comment by stbergmann** [comment]
(What this fix does not cover is live toggling of the theme after LO is started.  I'll leave that for a later fix.  From caolan's notes on IRC:  " fwiw, I think when you toggle the settings 'live' that we should get 'style-updated' signal called, i.e. signalStyleUpdated, which should call SalEvent::SettingsChanged, triggering GtkSalGraphics::updateSettings, and gtk-theme-name should be seen to change; maybe this is one of those things where the style-updated needs to be listened to level down on pEventWidget or something handwavy like that".)

**Comment by stbergmann** [comment]
seen fixed in flatpak rev 88e5b1b497dc
